.github/workflows/tests.yaml

```diff
@@ -82,19 +82,21 @@ jobs:
           ~/.composer/vendor/bin/php-coveralls -v
 
   functional-tests:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
     env:
       SELENIUM_SERVER_DOWNLOAD_URL: http://selenium-release.storage.googleapis.com/3.141/selenium-server-standalone-3.141.59.jar
 
     strategy:
       fail-fast: false
       matrix:
+        os: ['ubuntu-latest']
         browser: ['chrome', 'firefox']
         selenium-server: [true, false] # Whether to run via Selenium server or directly via browser driver
         w3c: [true] # Although all builds negotiate protocol by default, it implies W3C protocol for both Chromedriver and Geckodriver
         include:
           # Force OSS (JsonWire) protocol on ChromeDriver - to make sure we keep compatibility
-          - { browser: 'chrome', selenium-server: false, w3c: false }
+          - { browser: 'chrome', os: 'ubuntu-latest', selenium-server: false, w3c: false }
+          - { browser: 'safari', os: 'macos-latest', selenium-server: false }
```

Comment:
w3c must be implicitly set to true, otherwise strange things will happen (the scripts expects the value to be set to true or false)

Reply:
Based on https://developer.apple.com/documentation/webkit/macos_webdriver_commands_for_safari_12_and_later, since Safari 12, it's only W3C